### ISSUE TITLE
App crash with missing constructor on MvxImageView

### ISSUE THREAD

**Issue by auspex-nl** [MvvmCross/MvvmCross#1915]
It seems that the optional parameter added to some/most of the MvxImageView constructors will make the app crash when Activator.CreateInstance is called by MvxAndroidViewFactory due to it not being able to find the proper constructor on MvxImageView.

The constructors changed in this commit: https://github.com/MvvmCross/MvvmCross/commit/0b36189d8711d3bc5d666e6e6817c21bb6b36007#diff-f589bfb4a2af223a198d622ddfbb955b

This SO thread has some good information on Reflection & optional parameters:
https://stackoverflow.com/questions/2421994/invoking-methods-with-optional-parameters-through-reflection
Also this SO thread has some more specific information about Activator.CreateInstance:
https://stackoverflow.com/questions/11002523/activator-createinstance-with-optional-parameters

Imho, there's 2 ways to fix this, either change the mvximageview constructors (get rid of optional parametes) or change the way the instance is created in the MvxAndroidViewFactory.

### Steps to reproduce

1. Use the InternetMinute sample, use package version in the sample (4.0.0-beta2) or upgrade to 4.4.0
2. Build & verify that the app starts ok
3. Upgrade the mvvmcross packages to 5.0.1
4. Clean the solution (important to do this), build & start the app, it crashes on load of the HomeView

### Expected behavior
HomeView loads 

### Actual behavior
App crashes with unhandled exception:

```
06-05 19:18:43.712 I/MvxBind ( 7148):   1.49 Exception during creation of Mvx.MvxImageView from type MvvmCross.Binding.Droid.Views.MvxImageView - exception MissingMethodException: Constructor on type 'MvvmCross.Binding.Droid.Views.MvxImageView' not found.
06-05 19:18:43.712 I/MvxBind ( 7148): 	  at System.RuntimeType.CreateInstanceImpl (System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes, System.Threading.StackCrawlMark& stackMark) [0x00213] in <3fd174ff54b146228c505f23cf75ce71>:0 
06-05 19:18:43.712 I/MvxBind ( 7148):   at System.Activator.CreateInstance (System.Type type, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes) [0x000a1] in <3fd174ff54b146228c505f23cf75ce71>:0 
06-05 19:18:43.712 I/MvxBind ( 7148):   at System.Activator.CreateInstance (System.Type type, System.Object[] args) [0x00000] in <3fd174ff54b146228c505f23cf75ce71>:0 
06-05 19:18:43.712 I/MvxBind ( 7148):   at MvvmCross.Binding.Droid.Binders.MvxAndroidViewFactory.CreateView (Android.Views.View parent, System.String name, Android.Content.Context context, Android.Util.IAttributeSet attrs) [0x00019] in D:\git\MvvmCross2\MvvmCross\Binding\Droid\Binders\MvxAndroidViewFactory.cs:42 
[0:] MvxBind:Error:  1.49 Exception during creation of Mvx.MvxImageView from type MvvmCross.Binding.Droid.Views.MvxImageView - exception MissingMethodException: Constructor on type 'MvvmCross.Binding.Droid.Views.MvxImageView' not found.
	  at System.RuntimeType.CreateInstanceImpl (System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes, System.Threading.StackCrawlMark& stackMark) [0x00213] in <3fd174ff54b146228c505f23cf75ce71>:0 
  at System.Activator.CreateInstance (System.Type type, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes) [0x000a1] in <3fd174ff54b146228c505f23cf75ce71>:0 
  at System.Activator.CreateInstance (System.Type type, System.Object[] args) [0x00000] in <3fd174ff54b146228c505f23cf75ce71>:0 
  at MvvmCross.Binding.Droid.Binders.MvxAndroidViewFactory.CreateView (Android.Views.View parent, System.String name, Android.Content.Context context, Android.Util.IAttributeSet attrs) [0x00019] in D:\git\MvvmCross2\MvvmCross\Binding\Droid\Binders\MvxAndroidViewFactory.cs:42 
```

### Configuration

**Version:** 5.0.1

**Platform:** 
- [ ] iOS
- [x] Android
- [ ] WPF
- [ ] UWP

**Comment by Cheesebaron** [comment]
What version of Android is this?

**Comment by auspex-nl** [comment]
On both emulator (android 6.0) and real device (android 7.1)
target version of the android project was set to 5.0 in the sample, changed it to 7.1 to no avail
compile was set to 5.0 too, so changed that to 7.1 too

Linker is off

If I change to Release config (debugger off, Linker on (sdk assemblies only)), still the same.

**Comment by nmilcoff** [comment]
Would the issue be solved if we add a constructor like this?: 
```
public MvxImageView(Context context, IAttributeSet attrs, int defStyleAttr)
  :this(context, attrs, null)
```

**Comment by auspex-nl** [comment]
I don't think it will, as the calling code:

`var view = Activator.CreateInstance(viewType, context, attrs) as View;`

Will search for a constructor that matches something like: 

`public MvxImageView(Context context, IAttributeSet attrs)`

This constructor does not exist, due to the optional parameter. A quick fix would probably be to change this constructor:

```
        public MvxImageView(Context context, IAttributeSet attrs, Action afterImageChangeAction = null)
            : this(context, attrs, 0, afterImageChangeAction)
        { }
```

To these 2:

```
        public MvxImageView(Context context, IAttributeSet attrs)
            : this(context, attrs, 0)
        { }

        public MvxImageView(Context context, IAttributeSet attrs, Action afterImageChangeAction)
            : this(context, attrs, 0, afterImageChangeAction)
        { }
```

Whilst I'm personally not a fan of optional constructor parameters, I think the better fix would be to change the calling code that instantiates the object to pick the correct constructor.

**Comment by flyingxu** [comment]
I'm having the same issue after I upgraded MvvmCross from v4.4 to v5.0.2.

**Comment by Nick1611** [comment]
I have the same issue after upgraded from 4.4 to 5.0.3. No access to the view in code, simply declare in axml:

    <Mvx.MvxImageView
        android:id="@+id/listmapsnapshot"
        android:layout_width="100dp"
        android:layout_height="100dp"
        local:MvxBind="ImageUrl MapSnapShotPath" />

Error: 

[0:] MvxBind:Error:  2.85 Exception during creation of Mvx.MvxImageView from type MvvmCross.Binding.Droid.Views.MvxImageView - exception MissingMethodException: Constructor on type 'MvvmCross.Binding.Droid.Views.MvxImageView' not found.
	  at System.RuntimeType.CreateInstanceImpl (System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Object[] args, System.Globalization.CultureInfo culture, System.Object[] activationAttributes, System.Threading.StackCrawlMark& stackMark) [0x00197] in /Users/builder/data/lanes/4009/3a62f1ea/source/mono/mcs/class/referencesource/mscorlib/system/rttype.cs:5328 
InspectorDebugSession(2): HandleTargetEvent: UnhandledException

Thanks.

**Comment by Cheesebaron** [comment]
Yeah, sorry. This didn't make it into 5.0.3. Will have a look at fixing this ASAP.

**Comment by Cheesebaron** [comment]
@Nick1611 can you try latest CI build from https://www.myget.org/F/mvvmcross/api/v3/index.json and see if the issue persists?

**Comment by Nick1611** [comment]
which packages should I update for this fix?

**Comment by Cheesebaron** [comment]
In your case just MvvmCross.Binding. However, it will probably trickle down to all of the MvvmCross packages

**Comment by Nick1611** [comment]
@Cheesebaron You are right, all packages were updated. However, the error remains the same.

**Comment by auspex-nl** [comment]
I tried those CI bits on the example project that I mentioned in the original comment and for me the crash is gone.

**Comment by auspex-nl** [comment]
@Nick1611 Did you clean and/or rebuild the solution? otherwise the old binaries will be in use

**Comment by Nick1611** [comment]
My bad. I forgot to clean it this time. It is working fine after clean. Thanks

**Comment by Cheesebaron** [comment]
Excellent. Thank you very much for confirming this. This fix will definitely be out in 5.0.4.
